### ISSUE TITLE
Chemical Space Exploration

### ISSUE THREAD

**Issue by patrickhop** [deepchem/deepchem#671]
I wanted to open this one up to discussion.

DeepChem has built up a variety of high-quality models, some of which can actually get quite far with the limited public data out there. Today, we can take in an arbitrary molecule and estimate its properties reasonably well. What's missing imo is a good way of going from molecule to molecule, which is holding back our ability to do drug optimization here.

I'm not sure about autoencoders. It seems like a nice idea, but they don't really have much respect for synthesizability. I'm not sure hallucinating LSTMs are much better? In some ways I think matched molecular pairs might be the easiest option for now [http://rdkit.blogspot.com/2014/08/mmpa-on-chembl-herg-data-using-pandas.html].

Thoughts?

**Comment by rbharath** [comment]
I agree that a way of systematically exploring chemical space would be of high value (perhaps a deepchem 1.3 target? #669).

I like the idea of starting with matched molecular pairs. Simple chemoinformatics techniques are often the best baseline before trying more complicated ideas.

PRs welcome!

**Comment by rbharath** [comment]
I've labeled this issue a Research Question. Devising a deep learning algorithm that allows for systematic exploration of chemical space would likely constitute a publishable advance. If you're interested in working on this problem, please leave a comment here, and members of the steering committee might be interested in mentoring you.

**Comment by rbharath** [comment]
Closing this old issue as out of date; I think most people either use Enamine or generative AI methods to approach this now. See this DeepChem paper for some generative methods in DeepChem
